python/ray/autoscaler/command_runner.py

```diff
@@ -363,50 +663,27 @@ def run(self,
             "{}@{}".format(self.ssh_user, self.ssh_ip)
         ]
         if cmd:
-            final_cmd += _with_interactive(cmd)
+            if config["use_login_shells"]:
+                final_cmd += _with_interactive(cmd)
+            else:
+                final_cmd += [cmd]
             cli_logger.old_info(logger, "{}Running {}", self.log_prefix,
                                 " ".join(final_cmd))
         else:
             # We do this because `-o ControlMaster` causes the `-N` flag to
             # still create an interactive shell in some ssh versions.
             final_cmd.append(quote("while true; do sleep 86400; done"))
 
-        # todo: add a flag for this, we might
-        # wanna log commands with print sometimes
         cli_logger.verbose("Running `{}`", cf.bold(cmd))
         with cli_logger.indented():
             cli_logger.very_verbose("Full command is `{}`",
                                     cf.bold(" ".join(final_cmd)))
 
-        def start_process():
-            try:
-                if with_output:
-                    return self.process_runner.check_output(final_cmd)
-                else:
-                    self.process_runner.check_call(final_cmd)
-            except subprocess.CalledProcessError as e:
-                quoted_cmd = " ".join(final_cmd[:-1] + [quote(final_cmd[-1])])
-                if not cli_logger.old_style:
-                    raise ProcessRunnerError(
-                        "Command failed",
-                        "ssh_command_failed",
-                        code=e.returncode,
-                        command=quoted_cmd)
-
-                if exit_on_fail:
-                    raise click.ClickException(
-                        "Command failed: \n\n  {}\n".format(quoted_cmd)) \
-                        from None
-                else:
-                    raise click.ClickException(
-                        "SSH command Failed. See above for the output from the"
-                        " failure.") from None
-
         if cli_logger.verbosity > 0:
             with cli_logger.indented():
-                return start_process()
+                self._run_raw(final_cmd, with_output, exit_on_fail)
         else:
-            return start_process()
+            self._run_raw(final_cmd, with_output, exit_on_fail)
```

Comment:
actually, why is this called "run_raw"? what's the "opposite" function (run_unraw)? 

If there is none, would it make sense to just call it `_run_impl`?

Reply:
the "opposite" is `run_processed` i.e. `run()` since it adds `ssh -tt` and `_with_interactive` to the command before running it (if those things are enabled)

Reply:
Got it, can we call it `_run_impl`?